File: src/main/java/org/alham/alhamfirst/mapper/QuestMapper.java
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.dto.quest.QuestDTO;
import org.alham.alhamfirst.dto.stat.StatDTO;
import org.alham.alhamfirst.dto.todo.TodoDTO;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class QuestMapper {

    public QuestDTO createQuestDTO(TodoDTO todoDTO , StatDTO statDTO) {
        return QuestDTO.builder()
                .id(todoDTO.getId())
                .detail(todoDTO.getDetail())
                .completed(todoDTO.getCompleted())
                .statData(statDTO.getStatData())
                .build();
    }

    public List<QuestDTO> createQuestListDTO(List<TodoDTO> todoDTOList , List<StatDTO> statDTOList) {
        return todoDTOList.stream().map(todoDTO -> {
            StatDTO statDTO = statDTOList.stream().filter(stat -> stat.getTodoIdx().equals(todoDTO.getId())).findFirst().orElse(new StatDTO());
            return QuestDTO.builder()
                    .id(todoDTO.getId())
                    .detail(todoDTO.getDetail())
                    .completed(todoDTO.getCompleted())
                    .statData(statDTO.getStatData())
                    .build();
        }).toList();
    }

}

File: src/main/java/org/alham/alhamfirst/mapper/QuestMapper.kt
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.domain.document.QuestDocument
import org.alham.alhamfirst.domain.dto.quest.QuestDTO


class QuestMapper() {

     fun createQuestDTOFromDocument(questDocument: QuestDocument): QuestDTO {
        return QuestDTO(
            id = questDocument.id,
            userId = questDocument.userId,
            detail = questDocument.detail,
            completed = questDocument.completed,
            statData = questDocument.statData,
            startDate = questDocument.startDate,
            endDate = questDocument.endDate
        )
     }

    fun createDocumentFromDTO(questDTO: QuestDTO): QuestDocument {
        return QuestDocument(
            id = questDTO.id,
            userId = questDTO.userId,
            detail = questDTO.detail,
            completed = questDTO.completed,
            statData = questDTO.statData,
            startDate = questDTO.startDate,
            endDate = questDTO.endDate
        )
    }
}
